public/serviceworker/sw.js

```diff
@@ -0,0 +1,323 @@
+                                });
+                              } else {
+                                window.close();
+                                showCloseText();
+                              }
+                            });
+                            bc.postMessage({
+                              data: {
+                                instanceParams: instanceParams,
+                                hashParams: hashParams,
+                                queryParams: queryParams,
+                              },
+                              error: error,
+                            }).then(function () {
+                              setTimeout(function () {
+                                window.location.href = url.origin + location.search + location.hash;
+                              }, 5000);
+                            });
+                          }
+                        } catch (err) {
+                          console.error(err, "service worker error in redirect");
+                          bc && bc.close();
+                          window.close();
+                          showCloseText();
+                        }
+                      } else {
+                        // in preopen, awaiting redirect
+                        try {
+                          bc = new broadcastChannelLib.BroadcastChannel("preopen_channel_" + preopenInstanceId, broadcastChannelOptions);
+                          bc.onmessage = function (ev) {
+                            var { preopenInstanceId: oldId, payload, message } = ev.data;
+                            if (oldId === preopenInstanceId && payload && payload.url) {
+                              window.location.href = payload.url;
+                            } else if (oldId === preopenInstanceId && message === "setup_complete") {
+                              bc.postMessage({
+                                data: {
+                                  preopenInstanceId: preopenInstanceId,
+                                  message: "popup_loaded",
+                                },
+                              });
+                            }
+                            if (ev.error && ev.error !== "") {
+                              console.error(ev.error);
+                              bc.close();
+                            }
+                          };
+                        } catch (err) {
+                          console.error(err, "service worker error in preopen");
+                          bc && bc.close();
+                          window.close();
+                          showCloseText();
+                        }
+                      }
+                    </script>
+                  </body>
+                </html>
+                                        
+                ${''}
+                  `,
+						],
+						{ type: 'text/html' },
+					),
+				),
+			);
+		}
+	} catch (error) {
+		console.log('Hello');
```

Comment:
Hello to you as well!

Reply:
hahaha. This is the example service worker copied from the web3auth guide. It is def. not production ready.
I think it will also need adjustments as it hardcodes `localhost:3000`

https://web3auth.io/docs/content-hub/guides/tkey-mpc#setting-up-the-service-worker